### DISCUSSION TITLE
Strange graph behaviour

### DISCUSSION THREAD

**Discussion by stephencrowther** [RomRider/apexcharts-card#818] [Q&A]
Can anyone help with this strange quirk?

see pic attached why do I have the lines leading  in to wheee the data starts and come off the last data point?
![IMG_3371](https://github.com/user-attachments/assets/9088876e-d924-4678-a0b6-b4b35c63c85f)

**Comment by TheSerapher** [comment]
You need to to use the `extend_to: now` parameter to stop lines from progressing into the future. Be aware that mixing lines and columns will break the column width though. Since you only have lines it should be no issue.

**Comment by stephencrowther** [comment]
I reread the docs and figured this bit out in the end. The bit I couldn’t figure is why it’s doing that strange curve at the beginning 4 to 9 on graph. This is a solar inverter so those data points are all 0 until the inverter came online around 9.

I ended up adding the ‘use HA statistics’ setting in the end to stop this as I couldn’t figure out why.

**Comment by TheSerapher** [comment]
Excellent, glad you figured it out :) You could also add `group_by` to adjust the graph a bit and get rid of spikes. Any reason why the inverter comes online at 9 very suddenly? That's how it looks like on the graphs, instead of a slow buildup from `0` they just jump to whatever they are at `9`.

**Comment by stephencrowther** [comment]
No idea. The HA statistics look fine and start from 0 and gradually build as you would expect. But the apex chart statistics do this for some reason no idea why.

This graph is from template sensors from the raw inverter data because I needed to create sensors that were 0 instead of ‘unavailable’ so the template sensors stay at 0 until the inverter starts to give readings above 0.
